Source/Utils/WarpXUtil.cpp

```diff
@@ -444,9 +449,14 @@ void ReadBCParams ()
                 WarpX::particle_boundary_lo[idim] = ParticleBoundaryType::Periodic;
                 WarpX::particle_boundary_hi[idim] = ParticleBoundaryType::Periodic;
             }
-
         }
     }
+#ifdef WARPX_DIM_RZ
+    // Ensure code aborts if PEC is specified at r=0 for RZ
+    if ( WarpX::field_boundary_lo[0] == FieldBoundaryType::PEC) {
```

Comment:
To make this more general, this can be changed to `AMREX_ALWAYS_ASSERT_WITH_MESSAGE(WarpX::field_boundary_lo[0] == FieldBoundaryType::None, "Error: PEC boundary at r=0 must be 'none');`